newsroom/email.py

```diff
@@ -226,19 +231,23 @@ def get_language_template_name(template_name: str, language: str, extension: str
 TemplateKwargs = Dict[str, Any]
 
 
-# TODO-ASYNC: change this to use newsroom.users.model.UserResourceModel instead
+# TODO-ASYNC: change this to use newsroom.users.model.UserResourceModel only
 async def send_user_email(
-    user: User,
+    user: Union[User, "UserResourceModel"],
     template: str,
     template_kwargs: Optional[TemplateKwargs] = None,
     ignore_preferences=False,  # ignore user email preferences
     **kwargs: EmailKwargs,
 ) -> None:
     """Send an email to Newsroom user, respecting user's email preferences."""
+    if isinstance(user, BaseModel):
+        user = user.to_dict()
```

Comment:
`to_dict` is one of our methods and not from Pydantic, right? If that is the case, would be better to test for ResourceModel instead

Reply:
Indeed. I initially used BaseModel because of circular reference but then I did workaround it with TYPE_CHECKING flag. Fixed now.